File: server/router/content/content_short_video.go
package content

import "github.com/gin-gonic/gin"

type ShortVideoRouter struct{}

func (r *ShortVideoRouter) InitContentShortVideoRouter(private *gin.RouterGroup, public *gin.RouterGroup) {
	svPrivate := private.Group("contentShortVideo")
	{
		svPrivate.POST("createShortVideo", shortVideoApi.CreateShortVideo)
		svPrivate.PUT("updateShortVideo", shortVideoApi.UpdateShortVideo)
		svPrivate.DELETE("deleteShortVideo", shortVideoApi.DeleteShortVideo)
		svPrivate.GET("findShortVideo", shortVideoApi.FindShortVideo)
		svPrivate.GET("getShortVideoList", shortVideoApi.GetShortVideoList)
		svPrivate.POST("publishShortVideo", shortVideoApi.PublishShortVideo)
		svPrivate.POST("generateShortVideoScript", shortVideoApi.GenerateShortVideoScript)
		svPrivate.POST("createShortVideoWithAI", shortVideoApi.CreateShortVideoWithAI)
		svPrivate.POST("generateShortVideo", shortVideoApi.GenerateShortVideo)
		svPrivate.POST("regenerateShortVideo", shortVideoApi.RegenerateShortVideo)
	}

	svPublic := public.Group("public")
	{
		svPublic.GET("shortVideos", shortVideoApi.GetPublishedShortVideoList)
		svPublic.GET("shortVideo/:slug", shortVideoApi.GetPublishedShortVideoBySlug)
	}
}

File: web/src/api/contentShortVideo.js
import service from '@/utils/request'

export const getShortVideoList = (params) => {
  return service({
    url: '/contentShortVideo/getShortVideoList',
    method: 'get',
    params
  })
}

export const findShortVideo = (params) => {
  return service({
    url: '/contentShortVideo/findShortVideo',
    method: 'get',
    params
  })
}

export const createShortVideo = (data) => {
  return service({
    url: '/contentShortVideo/createShortVideo',
    method: 'post',
    data
  })
}

export const updateShortVideo = (data) => {
  return service({
    url: '/contentShortVideo/updateShortVideo',
    method: 'put',
    data
  })
}

export const deleteShortVideo = (data) => {
  return service({
    url: '/contentShortVideo/deleteShortVideo',
    method: 'delete',
    data
  })
}

export const publishShortVideo = (data) => {
  return service({
    url: '/contentShortVideo/publishShortVideo',
    method: 'post',
    data
  })
}

export const generateShortVideoScript = (data) => {
  return service({
    url: '/contentShortVideo/generateShortVideoScript',
    method: 'post',
    data,
    timeout: 600000
  })
}

export const createShortVideoWithAI = (data) => {
  return service({
    url: '/contentShortVideo/createShortVideoWithAI',
    method: 'post',
    data,
    timeout: 600000
  })
}

export const generateShortVideo = (data) => {
  return service({
    url: '/contentShortVideo/generateShortVideo',
    method: 'post',
    data,
    timeout: 600000
  })
}

export const regenerateShortVideo = (data) => {
  return service({
    url: '/contentShortVideo/regenerateShortVideo',
    method: 'post',
    data,
    timeout: 600000
  })
}

export const getPublishedShortVideos = (params) => {
  return service({
    url: '/public/shortVideos',
    method: 'get',
    params,
    donNotShowLoading: true
  })
}

export const getPublishedShortVideoBySlug = (slug) => {
  return service({
    url: `/public/shortVideo/${encodeURIComponent(slug)}`,
    method: 'get',
    donNotShowLoading: true
  })
}
